### ISSUE TITLE
Repeat print every 5 seconds

### ISSUE THREAD

**Issue by SherlocksHat** [scottbonline/sense#23]
This is an awesome script - thanks!

I want to feed this information into a file so that Splunk can play with it. How do I get active devices to display every 5 seconds?

For example:
('Active Devices:', u'Other, Always On')
('Active Devices:', u'Other, Fridge, Always On')
('Active Devices:', u'Other, Fridge, Always On')
('Active Devices:', u'Other, Oven, Always On')
('Active Devices:', u'Other, Kettle, Always On')

**Comment by kbickar** [comment]
Using the API as is, you could do it with the following code:
```python
from sense_api import Senseable
from time import sleep, time
sense = Senseable(api_timeout=20)
sense.rate_limit = 1
sense.authenticate(username, password)
last_call = 0
while True:
    if time() - last_call > 5:
        last_call = time()
        sense.get_realtime()
        print ("Active Devices:",", ".join(sense.active_devices))
    sleep(1)
```

An alternative would be to use a continuous stream of the websocket data, however, that would make it keep sending you data every 0.5 seconds so you'd be using 2-3 times the traffic.
I just added a get_realtime_stream function/generator that does just that if you want to go down that route.

**Comment by SherlocksHat** [comment]
Reads like what I'm looking for, just not sure how to actually run it. What command would I run to use the get_realtime_stream function for the updated script? or would I need to modify the script? Sorry I'm a bit of a newbie to Python.

**Comment by PopRe** [comment]
It'd be pretty much the same, but return a result every half second:

```
from sense_api import Senseable
from time import sleep, time
sense = Senseable(api_timeout=20)
sense.rate_limit = 1
sense.authenticate(username, password)
last_call = 0
for x in sense.get_realtime_stream():
    if time() - last_call > 5:
        last_call = time()
        sense.get_realtime()
        print ("Active Devices:",", ".join(sense.active_devices))
```

**Comment by kbickar** [comment]
It'd be pretty much the same, but return a result every half second:

```
from sense_api import Senseable
from time import sleep, time
sense = Senseable(api_timeout=20)
sense.rate_limit = 1
sense.authenticate(username, password)
last_call = 0
for x in sense.get_realtime_stream():
    if time() - last_call > 5:
        last_call = time()
        sense.get_realtime()
        print ("Active Devices:",", ".join(sense.active_devices))
```

**Comment by SherlocksHat** [comment]
ah brilliant - that works. 

Slight correction though which mattered for me being a newbie:

```
from sense_api import Senseable
from time import sleep, time
import pprint
import getpass

sense = Senseable(api_timeout=20)
sense.rate_limit = 1
username = raw_input("Please enter you Sense username (email address): ")
password = getpass.getpass("Please enter your Sense password: ")
sense.authenticate(username, password)
last_call = 0
while True:
    if time() - last_call > 5:
        last_call = time()
        sense.get_realtime()
        print ("Active Devices:",", ".join(sense.active_devices))
    sleep(1)
```

**Comment by bekirk** [comment]
This is awesome! I have not bought a sense yet but I didn’t want it if I couldn’t get access to my own data outside the app and web interface. Please let us know how the splunk goes I think it will unlock so much more capabilities.

**Comment by SherlocksHat** [comment]
The Sense python script is working fine. Unfortunately I've had a fair bit of trouble trying to get the data fed live into Splunk. Splunk uses its own version of python and doesn't include all the modules required by default to run the sense script (ie. websockets, etc). I've had to update the Splunk python to include the right modules and I have made some progress, but still not quite there yet.
